libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -2,18 +2,59 @@ package com.mapbox.navigation.route.offboard
 
 import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
+import android.content.Context
+import com.mapbox.api.directions.v5.models.DirectionsResponse
 import com.mapbox.geojson.Point
-import com.mapbox.navigation.base.route.Route
 import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.route.common.NavigationRoute
+import com.mapbox.navigation.route.common.extension.mapToRoute
+import com.mapbox.navigation.utils.exceptions.NavigationException
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 
 @MapboxNavigationModule(MapboxNavigationModuleType.OffboardRouter, skipConfiguration = true)
-class MapboxOffboardRouter : Router {
+class MapboxOffboardRouter(
+    private val context: Context,
+    private val mapboxToken: String
+) : Router {
 
-    override fun getRoute(origin: Point, waypoints: List<Point>, callback: (route: Route) -> Unit) {
-        TODO("not implemented")
+    private var navigationRoute: NavigationRoute? = null
+
+    override fun getRoute(
+        origin: Point,
+        waypoints: List<Point>?,
+        destination: Point,
+        callback: Router.RouteCallback
+    ) {
+        val builder = NavigationRoute
+            .builder(context)
+            .accessToken(mapboxToken)
+            .origin(origin)
+            .destination(destination)
+        waypoints?.forEach { builder.addWaypoint(it) }
+        navigationRoute = builder.build()
+        navigationRoute?.getRoute(object : Callback<DirectionsResponse> {
+            override fun onFailure(call: Call<DirectionsResponse>, t: Throwable) {
+                callback.onFailure(t)
+            }
+
+            override fun onResponse(
+                call: Call<DirectionsResponse>,
+                response: Response<DirectionsResponse>
+            ) {
+                val route = response.body()?.routes()?.firstOrNull()
+                if (response.isSuccessful && route != null) {
+                    callback.onRouteReady(route.mapToRoute())
```

Comment:
`DirectionsResponse` or any other wrapper should be returned here instead of only 1 route. Remember `DirectionsResponse` holds a list of potential `DirectionsRoute`s (alternatives) https://github.com/mapbox/mapbox-java/blob/9bd6158292374fc6dd5f7eba33e7586ca1d76177/services-directions/src/main/java/com/mapbox/api/directions/v5/models/DirectionsResponse.java#L87 + other data that might be needed.

Reply:
fixed